src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -333,7 +333,7 @@ public static (string assemblyName, string nameSpace, string shortClassName, str
 
             var nameSpace = "";
             var shortClassName = className;
-            var idx = fqn.LastIndexOf(".");
+            var idx = fqn.LastIndexOf('.');
```

Comment:
maybe we need to use `StringComparison.Ordinal` for all `IndexOf` here ? Eventually you can move this to separate PR